integration/insert_compat_test.go

```diff
@@ -124,18 +125,20 @@ func testInsertCompat(t *testing.T, testCases map[string]insertCompatTestCase) {
 						}
 
 						if targetErr != nil {
-							// Skip inserts that could not be performed due to Tigris schema validation.
-							var e mongo.BulkWriteException
-							if errors.As(targetErr, &e) && e.HasErrorCode(documentValidationFailureCode) {
-								setup.SkipForTigrisWithReason(t, targetErr.Error())
-							}
+							switch targetErr := targetErr.(type) { //nolint:errorlint // don't inspect error chain.
+							case mongo.WriteException:
+								// Skip inserts that could not be performed due to Tigris schema validation.
+								if targetErr.HasErrorCode(documentValidationFailureCode) {
```

Comment:
I thought previously it threw `mongo.BulkWriteException` but now it seems to be working with `mongo.WriteException`, not quite sure why. Maybe it was using errors.As and it unwrapped `BulkWriteException` previously.